server/src/main/kotlin/suwayomi/tachidesk/server/ServerSetup.kt

```diff
@@ -187,7 +227,10 @@ fun applicationSetup() {
     androidCompat.startApp(app)
 
     // Initialize NetworkHelper early
-    Injekt.get<NetworkHelper>()
+    Injekt.get<NetworkHelper>().userAgentFlow.onEach {
+        SystemProperties.set("http.agent", it)
+    }
```

Comment:
Need to launch the listener
```suggestion
    }.launchIn(GlobalScope)
```

Reply:
Thanks, I have never worked with StateFlow